### ISSUE TITLE
[Feature Request] AdMob: Rewarded Video Ads

### ISSUE THREAD

**Issue by svzi** [EddyVerbruggen/nativescript-plugin-firebase#467]
Do you have any plans to integrate "Rewarded Video Ads" (see https://firebase.google.com/docs/admob/android/quick-start and https://developers.google.com/admob/android/rewarded-video)?

I think that format is really interesting in some use cases and I would love to add that feature to my newest app.

**Comment by mayavi15** [comment]
@svzi Same here pal. I am developing a hobby app, in which user can get premium services for free if he watch a video. If i could get "Rewarded video" option in this plugin, it would be a great help.

**Comment by Acanguven** [comment]
I am also looking for this. It would be great.

**Comment by EddyVerbruggen** [comment]
Looks cool, and seems like a good fit for this plugin!

Until I find the time to look at it: if anyone has the time to take a look then I'd be happy to merge a PR. 

[Android doc](https://developers.google.com/admob/android/rewarded-video)
[iOS doc](https://developers.google.com/admob/ios/rewarded-video)

**Comment by marcosjunqueira** [comment]
+1

**Comment by cloudlegend** [comment]
I am also looking for this feature. It's been almost a year since the last comments. Any updates?

**Comment by umair-ameen** [comment]
Any updates on this this issue?

**Comment by umair-ameen** [comment]
@cloudlegend you found any other solution?

**Comment by cloudlegend** [comment]
@umair-ameen not yet, I am looking into the native API right now. I also open another issue in regards to the banner not loading after the initial load. Have u encountered this same issue? https://github.com/EddyVerbruggen/nativescript-admob/issues/15

Otherwise, I think we might have to access native API directly.

**Comment by umair-ameen** [comment]
@cloudlegend yup, if I try to load both ads at the same time, most of the it doesn't load the second call. So I am not loading them together at the same time.

**Comment by JCenas** [comment]
I had the same problem, i needed the reward video ad callback. 
So i forked this repo and implemented the missing feature.

You can check here the code https://github.com/4DevCoding/nativescript-admob
Fell free to improve or bring the changes to this repo :)

**Comment by muratcorlu** [comment]
Great @JCenas ! What about creating a Pull Request?

**Comment by JCenas** [comment]
@muratcorlu, It would be better when the ios implementation is finished.
And if someone could test it first would be great also :)

**Comment by muratcorlu** [comment]
@JCenas I'll try to check.

**Comment by sedobrengocce** [comment]
Hi, i started working on this feature
I've done the android part, but i've some problem with the IOS, someone can help me?

**Comment by anaszgh** [comment]
@sedobrengocce i was about to start with this when i saw your comment, how can i help to speed it up? 

@JCenas can't you just test it using the simulator? i may look into your fork and work on the IOS Part.

**Comment by Roar1827** [comment]
Anyone having any luck with this?

**Comment by JCenas** [comment]
@anaszgh I can't get ios simulator to windows or unix.

**Comment by NicholasSmilovic** [comment]
@anaszgh I believe the iOS implementation in @JCenas repo is just Interstitial ads still, JCenas mentioned they weren't able to do the iOS side.
https://github.com/4DevCoding/nativescript-admob/blob/37afee507f835fbe7193d51222bf38f3d8c2e7aa/src/admob.ios.js#L214

**Comment by NicholasSmilovic** [comment]
To help move things forward I tried to make a PR but it could use some help
https://github.com/EddyVerbruggen/nativescript-admob/pull/51

**Comment by EddyVerbruggen** [comment]
This has just landed in nativescript-admob. I'll add this here soon. https://github.com/EddyVerbruggen/nativescript-admob/pull/51
